src/modules/data/views/Messages/Messages.tsx

```diff
@@ -57,17 +70,74 @@ export const Messages: () => JSX.Element = () => {
     setViewMessage(history.location.state.viewMessage);
   }
 
+  const handleOpenFilter = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setFilterAnchor(event.currentTarget);
+  };
+
+  const handleAddFilter = (filter: string) => {
+    setActiveFilters((activeFilters) => [...activeFilters, filter]);
+  };
+
+  useEffect(() => {
+    // set filters if they are present in the URL
+    if (filterQuery.length !== 0) {
+      setActiveFilters(filterQuery as string[]);
+    }
+  }, [setActiveFilters, filterQuery]);
+
+  useEffect(() => {
+    //set query param state
+    setFilterQuery(activeFilters);
+    if (activeFilters.length === 0) {
+      setFilterString('');
+      return;
+    }
+
+    setFilterString(`&${activeFilters.join('&')}`);
+  }, [activeFilters, setFilterQuery]);
+
+  const filterFields = [
+    'author',
+    'batch',
+    'cid',
+    'confirmed',
+    'created',
+    'group',
+    'hash',
+    'id',
+    'key',
+    'local',
+    'namespace',
+    'pending',
+    'pins',
+    'rejected',
```

Comment:
local, pending, and rejected were rolled together as "state" in https://github.com/hyperledger/firefly/pull/296

Reply:
good catch, thanks